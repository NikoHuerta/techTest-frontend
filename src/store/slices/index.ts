export { default as taskReducer } from './taskSlice';
export { default as navigationReducer } from './navigationSlice';
export * from './navigationSlice';
export * from './taskSlice';