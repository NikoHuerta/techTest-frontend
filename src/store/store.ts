import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import { taskReducer, navigationReducer } from './slices';

export const store = configureStore({
  reducer: {
    task: taskReducer,
    navigation: navigationReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
