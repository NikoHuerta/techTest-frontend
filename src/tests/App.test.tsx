import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../store';
import { App } from '../App';


describe( 'Pruebas en App.tsx', () => {

  test( 'Debe renderizar el componente App correctamente', () => {
    const wrapper = render( <Provider store={ store }><App /></Provider> );
    expect( wrapper ).toMatchSnapshot();
  } );
  
});
