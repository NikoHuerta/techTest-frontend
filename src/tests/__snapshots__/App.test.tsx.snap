// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en App.tsx Debe renderizar el componente App correctamente 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1kg46n1-MuiSvgIcon-root"
                data-testid="AdbIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 16c0 3.87 3.13 7 7 7s7-3.13 7-7v-4H5v4zM16.12 4.37l2.1-2.1-.82-.83-2.3 2.31C14.16 3.28 13.12 3 12 3s-2.16.28-3.09.75L6.6 1.44l-.82.83 2.1 2.1C6.14 5.64 5 7.68 5 10v1h14v-1c0-2.32-1.14-4.36-2.88-5.63zM9 9c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm6 0c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1z"
                />
              </svg>
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
                value="Inicio"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1uwli8v-MuiTypography-root"
                >
                  TAREAS APP
                </h6>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <div
                class="MuiBox-root css-mfiwgy"
              >
                <button
                  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1q39md6-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                  value="Ver Tareas"
                >
                  Ver Tareas
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1q39md6-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                  value="Nueva Tarea"
                >
                  Nueva Tarea
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </header>
      </nav>
      <main
        style="margin: 80px auto; max-width: 1440px; padding: 0px 30px;"
      >
        <h1>
          Home Screen
        </h1>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-8paf5q-MuiTypography-root"
        >
          App encargada de manejar tareas.
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-160tic2-MuiTypography-root"
        >
          Para ver las tareas, haga click en el boton "Ver Tareas".
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-1oxen2w-MuiTypography-root"
        >
          Para agregar una nueva tarea, haga click en el boton "Nueva Tarea".
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-1oxen2w-MuiTypography-root"
        >
          En la visualizacion usted podra Editar y Borrar una tarea respectiva.
        </h6>
      </main>
    </div>
  </body>,
  "container": <div>
    <nav>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1kg46n1-MuiSvgIcon-root"
              data-testid="AdbIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5 16c0 3.87 3.13 7 7 7s7-3.13 7-7v-4H5v4zM16.12 4.37l2.1-2.1-.82-.83-2.3 2.31C14.16 3.28 13.12 3 12 3s-2.16.28-3.09.75L6.6 1.44l-.82.83 2.1 2.1C6.14 5.64 5 7.68 5 10v1h14v-1c0-2.32-1.14-4.36-2.88-5.63zM9 9c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm6 0c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1z"
              />
            </svg>
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
              value="Inicio"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1uwli8v-MuiTypography-root"
              >
                TAREAS APP
              </h6>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <div
              class="MuiBox-root css-mfiwgy"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1q39md6-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
                value="Ver Tareas"
              >
                Ver Tareas
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1q39md6-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
                value="Nueva Tarea"
              >
                Nueva Tarea
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </header>
    </nav>
    <main
      style="margin: 80px auto; max-width: 1440px; padding: 0px 30px;"
    >
      <h1>
        Home Screen
      </h1>
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-8paf5q-MuiTypography-root"
      >
        App encargada de manejar tareas.
      </h6>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-160tic2-MuiTypography-root"
      >
        Para ver las tareas, haga click en el boton "Ver Tareas".
      </h6>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-1oxen2w-MuiTypography-root"
      >
        Para agregar una nueva tarea, haga click en el boton "Nueva Tarea".
      </h6>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-1oxen2w-MuiTypography-root"
      >
        En la visualizacion usted podra Editar y Borrar una tarea respectiva.
      </h6>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
